### PR TITLE
Br68 test module for data handling

### PR DESCRIPTION
**Pull request by zengxyu**

### CODE REVIEW

File: pytest/data_handling/postprocessors/test_postprocessors.py [hunk 1]
```diff
@@ -1,6 +1,63 @@
...
+        return sample_position
+
+    @pytest.fixture
+    def custom_encoders(self):
+        custom_encoders = {"categorical2": CategoricalEncoder}
+        return custom_encoders
+
+    @pytest.mark.parametrize('sequential', [True, False])
+    def test_postprocess(self, iterators, sample_position, sample, custom_encoders, sequential):
+        feature_encoding_configs = [{"feature_type": "categorical", "feature_names": [0, 1, 2], "train_split": "train"}]
```
Comment (le1nux) [inline pytest/data_handling/postprocessors/test_postprocessors.py:40]: we should also test for the "continuuous" feature type

File: pytest/data_handling/test_post_processed_dataset_iterator.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def post_processor(self):
+        postprocessor = OneHotEncodedTargetPostProcessor(target_vector_size=10, target_position=1)
+        return postprocessor
+
+    def test_len(self, iterator, post_processor):
+        postprocess_dataset_iterator = PostProcessedDatasetIterator(dataset_iterator=iterator,
+                                                                    post_processor=post_processor)
+        assert len(postprocess_dataset_iterator) == len(iterator)
+        assert postprocess_dataset_iterator.underlying_iterators[0] == iterator
+        item = postprocess_dataset_iterator.__getitem__(1)
```
Comment (le1nux) [inline pytest/data_handling/test_post_processed_dataset_iterator.py:30]: better: postprocess_dataset_iterator[1]

**Review by le1nux — CHANGES_REQUESTED** [review]
so far so good. :) Left a few comments that should be addressed. Thanks :+1:

File: pytest/blueprints/constructables/test_higher_order_iterator_constructables.py [hunk 3]
```diff
@@ -81,6 +85,49 @@ def test_constructable(self, informed_iterators):
...
+        constructable = InMemoryDatasetIteratorConstructable(component_identifier="memory_component",
+                                                             requirements=requirements)
+        iterators = constructable.construct()
+        iterator_train, iterator_test = iterators["train"], iterators["test"]
+        sample, target, tag = iterator_train[0]
+        assert list(sample.shape) == [28, 28]
+        assert isinstance(target, int)
+
+        assert isinstance(iterator_train, InformedDatasetIterator)
+        assert isinstance(iterator_test, InformedDatasetIterator)
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/blueprints/constructables/test_higher_order_iterator_constructables.py:100]: done

File: pytest/blueprints/constructables/test_higher_order_iterator_constructables.py [hunk 4]
```diff
@@ -81,6 +85,49 @@ def test_constructable(self, informed_iterators):
...
+        shuffled_sample, shuffled_target, shuffled_tag = shuffled_iterators["train"][0]
+
+        assert list(sample.shape) == [28, 28]
+        assert isinstance(target, int)
+
+        assert isinstance(shuffled_iterators["train"], InformedDatasetIterator)
+        assert isinstance(shuffled_iterators["test"], InformedDatasetIterator)
+
+        # assert if shuffled_sample != sample
+        assert (shuffled_sample != sample).any()
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/blueprints/constructables/test_higher_order_iterator_constructables.py:128]: done

File: pytest/blueprints/constructables/test_higher_order_iterator_constructables.py [hunk 5]
```diff
@@ -152,6 +199,25 @@ def test_constructable(self, informed_iterators):
...
+                                                                  applicable_splits=["train"])
+        iterators = constructable.construct()
+        _, label, _ = informed_iterators["train"][0]
+        iterator_train_encoded = iterators["train"]
+        sample, target, tag = iterator_train_encoded[0]
+        assert list(sample.shape) == [28, 28]
+        assert len(target) == target_vector_size
+        assert target[int(label)] == 1
+
+        assert isinstance(iterator_train_encoded, InformedDatasetIteratorIF)
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/blueprints/constructables/test_higher_order_iterator_constructables.py:218]: done

File: pytest/blueprints/constructables/test_registry_contructable.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    def test_constructable(self, mocked_nn_class, seed, layer_config, prediction_publication_key):
+        constructable = ModelRegistryConstructable()
+        model_registry = constructable.construct()
+        model_registry.add_class("mocked_nn", mocked_nn_class)
+
+        assert isinstance(model_registry, ClassRegistry)
+        assert model_registry._store["mocked_nn"] == mocked_nn_class
+
+        # assert if it is able to get a model instance using model_registry
+        model = model_registry.get_instance("mocked_nn", seed=seed, layer_config=layer_config,
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/blueprints/constructables/test_registry_contructable.py:50]: done

File: pytest/data_handling/postprocessors/test_feature_encoder.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        return values
+
+    @pytest.fixture
+    def sample(self):
+        return np.array([1])
+
+    def test_fit_and_transform(self, encoder, values, sample):
+        encoder.fit(values)
+        transformed_sample = encoder.transform(sample)
+        assert transformed_sample[0][sample[0]] == 1
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/data_handling/postprocessors/test_feature_encoder.py:25]: done, check if the sum of the vector is 1

File: pytest/data_handling/postprocessors/test_feature_encoder.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+    @pytest.fixture
+    def sample(self):
+        return [0.5]
+
+    def test_fit_and_transform(self, encoder, values, sample):
+        encoder.fit(values)
+        transformed_sample = encoder.transform(sample)
+        assert transformed_sample[0][0] < 1
+        assert encoder.get_output_size() == 1
```
[the remark this answers is not in the document]
Comment (zengxyu) [reply pytest/data_handling/postprocessors/test_feature_encoder.py:48]: done

File: pytest/data_handling/test_post_processed_dataset_iterator.py (hunk 2 above)
Comment (zengxyu) [reply pytest/data_handling/test_post_processed_dataset_iterator.py:30]: done

File: pytest/data_handling/postprocessors/test_postprocessors.py (hunk 1 above)
Comment (zengxyu) [reply pytest/data_handling/postprocessors/test_postprocessors.py:40]: done

File: pytest/blueprints/constructables/test_higher_order_iterator_constructables.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (le1nux) [reply pytest/blueprints/constructables/test_higher_order_iterator_constructables.py]: Where do I find the functionality tests?

**Pull request merged**